velox/dwio/parquet/reader/ParquetData.cpp

```diff
@@ -140,4 +212,48 @@ std::pair<int64_t, int64_t> ParquetData::getRowGroupRegion(
   return {fileOffset, length};
 }
 
+std::shared_ptr<BloomFilter> ParquetData::getBloomFilter(
+    dwio::common::BufferedInput& bufferedInput,
+    const uint32_t rowGroupId) {
+  auto columnBloomFilterIter = columnBloomFilterMap_.find(rowGroupId);
+  if (columnBloomFilterIter != columnBloomFilterMap_.end()) {
+    return columnBloomFilterIter->second;
+  }
+
+  VELOX_CHECK_LT(
+      rowGroupId,
+      fileMetaDataPtr_.numRowGroups(),
+      "Invalid row group ordinal: {}",
+      rowGroupId);
+
+  auto rowGroup = fileMetaDataPtr_.rowGroup(rowGroupId);
+  auto colChunk = rowGroup.columnChunk(type_->column());
+  VELOX_CHECK(
+      !colChunk.hasCryptoMetadata(), "Cannot read encrypted bloom filter yet");
+
+  if (!colChunk.hasBloomFilterOffset()) {
+    return nullptr;
+  }
+  auto bloomFilterOffset = colChunk.bloomFilterOffset();
+  auto fileSize = bufferedInput.getInputStream()->getLength();
+  VELOX_CHECK_GT(
+      fileSize,
+      bloomFilterOffset,
+      "file size {} less or equal than bloom offset {}",
+      fileSize,
+      bloomFilterOffset);
+
+  auto inputStream = bufferedInput.read(
```

Comment:
This is wrong time to do IO here.  This must be scheduled in `StructColumnReader::loadRowGroup` and the `SeekableInputStream` should be stored as a field in `ParquetData`, instead of passing it extra parameter in `filterRowGroups`.

Reply:
`StructColumnReader::loadRowGroup` is called after `filterRowGroups` where bloom filter checks happen. Made it a member of `ParquetData` and initializing it at the beginning of the `filterRowGroups` method.